inst/templates/ad_adsl.R

```diff
@@ -189,7 +210,19 @@ adsl <- adsl %>%
     start_date = TRTEDT,
     end_date = DTHDT,
     add_one = FALSE
-  )
+  ) %>%
+  # Cause of Death and Traceability Variables
+  derive_var_dthcaus(
+    src_ae, src_ds,
+    source_datasets = list(ae = ae, ds = ds_ext)
```

Comment:
Do we want to use the new functions that @jeffreyad just made in https://github.com/pharmaverse/admiral/pull/2236 since we are superseding this function?

Reply:
Really good point, thanks - it had slipped my mind. I will thus wait until #2236 is merged and then use that instead.

Reply:
Here is some example code for using `derive_vars_extreme_event()`

```
adsl <- adsl %>%
  derive_vars_extreme_event(
    by_vars = exprs(STUDYID, USUBJID),
    events = list(
      event(
        dataset_name = "ae",
        condition = AEOUT == "FATAL",
        set_values_to = exprs(DTHCAUS = AEDECOD),
      ),
      event(
        dataset_name = "ds",
        condition = DSDECOD == "DEATH" & grepl("DEATH DUE TO", DSTERM),
        set_values_to = exprs(DTHCAUS = DSTERM),
      )
    ),
    source_datasets = list(ae = ae, ds = ds),
    tmp_event_nr_var = event_nr,
    order = exprs(event_nr),
    mode = "first",
    new_vars = exprs(DTHCAUS = DTHCAUS)
  )
```